packages/mcp-auth/src/index.ts

```diff
@@ -97,19 +125,46 @@ export type BearerAuthJwtConfig = {
  * ```
  */
 export class MCPAuth {
-  constructor(public readonly config: MCPAuthConfig) {
-    const result = validateServerConfig(config.server);
+  /**
+   * Deduplicated list of authorization servers, combined from:
+   * 1. The configured auth server metadata (if provided)
+   * 2. Authorization servers defined in protectedResource metadata
+   *
+   * These auth servers are used for dynamic JWT validation, where the issuer
+   * in the JWT token is matched against available auth servers to find the
+   * correct JWKS endpoint for token verification.
+   */
+  private readonly availableAuthServers: AuthServerConfig[];
 
-    if (!result.isValid) {
+  constructor(public readonly config: MCPAuthConfig) {
+    const { server, protectedResource } = config;
+    if (!server && !protectedResource) {
       throw new MCPAuthAuthServerError('invalid_server_config', {
-        ...result,
+        cause: 'No authorization server or protected resource metadata is provided.',
       });
     }
 
-    if (result.warnings.length > 0) {
-      console.warn(
-        `The authorization server configuration has warnings:\n\n  - ${result.warnings.map(({ description }) => description).join('\n  - ')}\n`
-      );
+    this.availableAuthServers = deduplicateAuthServers(
+      [server, ...(protectedResource?.metadata.authorizationServers ?? [])].filter(
```

Comment:
for dups, we just throw an error instead? although we can provide backwards compatibility for `server`, but will it be better to force devs to remove the use of `server` once they are trying to use `protectedResource`?

Reply:
Good call, updated.